### DISCUSSION TITLE
Migrate from Newtonsoft.Json to System.Text.Json

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6135] [Serenity Ideas]
Why we strict to `Newtonsoft.Json` where built in features are available. and `System.Text.Json` claimed to be faster and secure than `Newtonsoft.Json`.

[A comparison of Newtonsoft.Json and System.Text.Json](https://inspiration.nlogic.ca/en/a-comparison-of-newtonsoft.json-and-system.text.json)

[Comparing the Performance of System.Text.Json and Newtonsoft.Json in .NET Core](https://erickmccollum.com/2020/03/01/Serialization-With-SystemTextJson.html)

[How to migrate from Newtonsoft.Json to System.Text.Json](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-6-0)

I am just wondering and sharing my thoughts and want to know your thoughts on this.
Thanks.

**Comment by VictorTomaili** [comment]
Because of JSON customization which doesn't have support by System.Text.Json. Like JsonRowConverter and JsonSafeCriteriaConverter etc.

**Reply by dfaruque** [reply]
It seems that this is also possible.

[How to write custom converters for JSON serialization (marshalling) in .NET](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-6-0)

**Reply by VictorTomaili** [reply]
A few months ago, we had some restrictions which don't allow us to use it. I don't remember them now. But nowadays, maybe it can be used as required. Probably we will evaluate System.Text.Json for the ".net 6" version of Serenity.
